django/lego/dns.py

```diff
@@ -0,0 +1,47 @@
+# Copyright 2024 Canonical Ltd.
+# See LICENSE file for licensing details.
+"""DNS utiilities."""
+
+from pathlib import Path
+from tempfile import TemporaryDirectory
+
+from git import Git, Repo
+
+from .settings import DNS_REPOSITORY_URL, SSH_IDENTITY_FILE
+
+REPOSITORY_BASE_URL = "@".join(DNS_REPOSITORY_URL.split("@")[:-1])
+REPOSITORY_BRANCH = DNS_REPOSITORY_URL.split("@")[-1]
```

Comment:
What does this do please? Is it removing some suffix?

Reply:
That splits a repository URL like https://github.com/canonical/httprequest-lego-provider.git@main into the base URL and the branch

Reply:
Hmm, the `join` on line 12 seems redundant then?

Reply:
I'm splitting by `@`, discarding the last element and joining the rest of the elements again, in case there's a `@` somewhere

Reply:
I see, you can just do an [rsplit](https://docs.python.org/3/library/stdtypes.html#str.rsplit)